scipy/optimize/_optimize.py

```diff
@@ -1064,8 +1070,9 @@ def g(w, func, x0, v, *args):
         raise ValueError("{} is not a valid string for "
                          "``direction`` argument".format(direction))
 
-    return sqrt(sum((analytical_grad -
-                     approx_fprime(vars, _func, step, *_args))**2))
+    return np.sqrt(np.sum(np.abs(
+        (analytical_grad - approx_fprime(vars, _func, step, *_args))**2
+    )))
```

Comment:
If we have a vector, don't we want to have the possibility to access the whole error vector?

Reply:
From the existing docs:

> err : float
>     The square root of the sum of squares (i.e., the 2-norm) of the
>     difference between ``grad(x0, *args)`` and the finite difference
>     approximation of `grad` using func at the points `x0`.

The `check_grad` function is supposed to return a float by reducing the `(n,)` derivative array to a single value. This behaviour is unchanged, except it also does it for an (m, n) array.

Reply:
I was not clear, my question was more: should we allow the user to request a vector? Like some metric in scikit-learn were you can request a scalar or vector (and for some you can also specify how the scalar is computed along the vector)